python/cudf/cudf/core/column/column.py

```diff
@@ -2900,12 +2918,25 @@ def as_column(
                 ):
                     # Conversion to arrow converts IntervalDtype to StructDtype
                     dtype = CategoricalDtype.from_pandas(arbitrary.dtype)
-            return as_column(
+            result = as_column(
                 pa.array(arbitrary, from_pandas=True),
                 nan_as_null=nan_as_null,
                 dtype=dtype,
                 length=length,
             )
+            if (
+                cudf.get_option("mode.pandas_compatible")
+                and isinstance(arbitrary.dtype, pd.CategoricalDtype)
+                and is_pandas_nullable_extension_dtype(
+                    arbitrary.dtype.categories.dtype
+                )
+                and dtype is None
```

Comment:
This should probably be in the `if isinstance(arbitrary.dtype, pd.CategoricalDtype):` branch above to override `dtype` if it's none (like in the `isinstance(arbitrary.dtype.categories.dtype, pd.IntervalDtype)`) condition above

Reply:
If we move it to the location you suggested, `self._set_categories` get's triggered and that inturn triggers `merge`. Currently `merge` errors with different pandas nullable extension dtyps combinations. I'm aiming to improve this in `25.10`. I left a todo here.